### PR TITLE
Competition-Detector-Fix: Jira

### PR DESCRIPTION
**Pull request by ankushgoel27**
Purpose:

JIratoken v2 is not in the pkg/engine/defaults.go so added that
Dont see the use of prefix regex behind domain name and email regex so removed that.
Should i also remove the prefix regex from the token pat? makes sense as the domain regex already contains the atlassian keyword.
Fixed the token regex for jiratokenv2 . Boundary at the end wont work as the regex contains `=` char so the regex wont capture the entire token string.

jiratoken
jiratokenv2

Tests work but the jiratoken detector is not working. tested with other v2 detectors and they are also not working. i think the versioner interface is broken. 

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/jiratoken/jiratoken.go [hunk 1]
```diff
@@ -73,9 +73,9 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 					Raw:          []byte(resToken),
 					RawV2:        []byte(fmt.Sprintf("%s:%s:%s", resEmail, resToken, resDomain)),
 				}
-				s1.ExtraData = map[string]string{
-					"rotation_guide": "https://howtorotate.com/docs/tutorials/atlassian/",
-				}
+				//s1.ExtraData = map[string]string{
```
Comment (zricethezav) [inline pkg/detectors/jiratoken/jiratoken.go:76]: @ankushgoel27 any reason why this is commented out?

Comment (ankushgoel27) [reply pkg/detectors/jiratoken/jiratoken.go]: i initially commented them out due to go test command failing for not ignoring extradata in the test file. I added that and now the test's pass.

i believe this is ready for merge. Both detectors work and both tests pass.

File: pkg/detectors/jiratoken/jiratoken_test.go [hunk 2]
```diff
@@ -138,7 +138,7 @@ func TestJiraToken_FromChunk(t *testing.T) {
 					t.Fatalf("wantVerificationError = %v, verification error = %v", tt.wantVerificationErr, got[i].VerificationError)
 				}
 			}
-			ignoreOpts := cmpopts.IgnoreFields(detectors.Result{}, "Raw", "RawV2", "VerificationError")
+			ignoreOpts := cmpopts.IgnoreFields(detectors.Result{}, "Raw", "RawV2", "ExtraData", "VerificationError")
```
Comment (rosecodym) [inline pkg/detectors/jiratoken/jiratoken_test.go:141]: It looks like the extra data that the detector is adding isn't dependent on the secret, so you can just add it to the test cases. Then you won't have to ignore this field in the comparison.

**Review by chiraggl — APPROVED** [review]

**Comment by ankushgoel27** [comment]
closing this PR as Jira V2 is already implemented and fixed and working in the latest version

**Pull request closed without merging**